sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVExprUtils.scala

```diff
@@ -79,4 +82,48 @@ object CSVExprUtils {
         throw new IllegalArgumentException(s"Delimiter cannot be more than one character: $str")
     }
   }
+
+  /**
+   * Helper method that converts string representation of a character sequence to actual
+   * delimiter characters. The input is processed in "chunks", and each chunk is converted
+   * by calling [[CSVExprUtils.toChar()]].  A chunk is either:
+   * <ul>
+   *   <li>a backslash followed by another character</li>
+   *   <li>a non-backslash character by itself</li>
+   * </ul>
+   * , in that order of precedence. The result of the converting all chunks is returned as
+   * a [[String]].
+   *
+   * <br/><br/>Examples:
+   * <ul><li>`\t` will result in a single tab character as the separator (same as before)
```

Comment:
Sorry if I am rushing to read but why do we need this escaping logic? can't we just set the option as `"\t"` instead of `"""\t"""`?

or .. is something Univocity expects? If this is the case, it would be easier to review if you link Univocity side's doc.

Reply:
Yeah it's a fair point @jeff303 ; won't this already be unescaped by virtue of being a string literal? If I pass `"\\"` then the string literal is a single backslash, by the time it gets here. That would yield an error, not specify a single backslash as delimiter.

Reply:
This same question also applies to the existing method (`toChar`), right? I'm attempting to adhere to the same semantics that it's using.  My assumption was that at some level, this input might come from some context that can't directly specify a tab character (web UI, another language, etc.).

Reply:
It looks like the existing logic was put into place under #22379.

Reply:
Yeah you're right I'm also questioning why this method is there to parse the delimiter at all. Seems like the result of passing `"\\"` would be surprising. It existed before that change, even. @MaxGekk do you recall anything about this part? why does the string need further unescaping?

Well.. we could say that's a separate question and just leave this issue aside here entirely.

Reply:
Let me suggest discussing this point at https://github.com/apache/spark/pull/26027#discussion_r332182154 below